### ISSUE TITLE
WashingtonDC crashes when you try to play Crazy Box mode in Crazy Taxi

### ISSUE THREAD

**Issue by snickerbockers** [washingtondc-emu/washingtondc#31]
This happens when you try to select a level after inputting your name.  The cab driver's bio appears (which seems to have a missing polygon, btw) and then WashingtonDC crashes with an ERROR_INTEGRITY.

```
Total elapsed time: 473 seconds and 342306338 nanoseconds
2630573159 SH4 CPU cycles executed
Performance is 32.778551 MHz (16.389275%)
ERROR: something that *should* be impossible just happened
[sh4_reg_mmucr] = 0
[sh4_reg_tea] = 0
[sh4_reg_ttb] = 0
[sh4_reg_ptea] = 0
[sh4_reg_ptel] = 0
[sh4_reg_pteh] = 0
[sh4_reg_qacr1] = 10
[sh4_reg_qacr0] = 10
[sh4_reg_ccr] = 90d
[sh4_reg_r15] = c2fb3a4
[sh4_reg_r14] = 1
[sh4_reg_r13] = 3
[sh4_reg_r12] = 4
[sh4_reg_r11] = 0
[sh4_reg_r10] = 4
[sh4_reg_r9] = a05f8060
[sh4_reg_r8] = a05f8008
[sh4_reg_r7_bank1] = a05f8144
[sh4_reg_r6_bank1] = 4
[sh4_reg_r5_bank1] = 1
[sh4_reg_r4_bank1] = 20
[sh4_reg_r3_bank1] = c2f669c
[sh4_reg_r2_bank1] = ffffffff
[sh4_reg_r1_bank1] = a05f8014
[sh4_reg_r0_bank1] = c2e7e80
[sh4_reg_r7_bank0] = 0
[sh4_reg_r6_bank0] = 0
[sh4_reg_r5_bank0] = 0
[sh4_reg_r4_bank0] = 0
[sh4_reg_r3_bank0] = 0
[sh4_reg_r2_bank0] = 0
[sh4_reg_r1_bank0] = 0
[sh4_reg_r0_bank0] = 0
[sh4_reg_fpul] = 0
[sh4_reg_fpscr] = 40001
[sh4_reg_pr] = c16a284
[sh4_reg_macl] = 68
[sh4_reg_mach] = 0
[sh4_reg_dbr] = 8c000010
[sh4_reg_sgr] = c00f3d0
[sh4_reg_vbr] = 8c00f400
[sh4_reg_gbr] = c2b5060
[sh4_reg_spc] = c15841a
[sh4_reg_pc] = c169ec0
[sh4_reg_ssr] = 60000000
[sh4_reg_sr] = 600000f0
[function] = "pvr2_tex_cache_read"
[file] = "/home/jay/programs/washingtondc/src/hw/pvr2/pvr2_tex_cache.c"
[line] = 422
Aborted (core dumped)
```

**Comment by snickerbockers** [comment]
Looks like either it's crashing on a failed allocation, or one of the unimplemented pixel formats got used.  From src/hw/pvr2/pvr2_tex_cache.c (the ERROR_INTEGRITY seen in this bug is the one raised if tex_dat is NULL):

```
    unsigned tex_w = 1 << meta->w_shift, tex_h = 1 << meta->h_shift;

    // TODO: better error-handling
    if ((ADDR_TEX64_LAST - ADDR_TEX64_FIRST + 1) <=
        (meta->addr_last - meta->addr_first + 1)) {
        abort();
    }

    size_t n_bytes = sizeof(uint8_t) *
        (1 << meta->w_shift) * (1 << meta->h_shift) *
        pixel_sizes[meta->tex_fmt];

    void *tex_dat = NULL;
    if (n_bytes)
        tex_dat = malloc(n_bytes);

    if (!tex_dat)
        RAISE_ERROR(ERROR_INTEGRITY);
```

**Comment by snickerbockers** [comment]
The root-cause is that TEX_CTRL_PIX_FMT_4_BPP_PAL is unimplemented.

**Comment by snickerbockers** [comment]
This issue has been fixed by commit 61cfb3f2fad55cd08a0740b88e465871cd0ba1c1.
